source/connect/connection-options.txt

```diff
@@ -0,0 +1,328 @@
+
+.. literalinclude:: /includes/connect/connection-options.php
+   :language: php
+   :copyable: true
+   :start-after: // start-client-options 
+   :end-before: // end-client-options
+   :emphasize-lines: 5-8, 11
+
+.. note::
+   
+   If you specify an option in both the ``$uriOptions`` parameter and in the connection
+   URI, the value in ``$uriOptions`` takes precedence.
+
+Connection URI Options
+----------------------
+
+The following sections describe the options that you can set for your connection to
+MongoDB. Each connection option links to its corresponding
+entry in the {+mdb-server+} manual.
+
+.. important:: Percent-Encoding
+
+   If the value of a connection option contains special characters, you must
+   :wikipedia:`percent-encode <Percent-encoding>` the value before including it
+   in the connection URI. You can use the ``rawurlencode()`` method to encode 
+   these values according to the URI syntax specified in RFC 3986.
+
+   Don't percent-encode connection options when including them in the
+   ``$uriOptions`` parameter.
+
+   To learn more, see the following resources:
+
+   - `RFC 3986 <http://www.faqs.org/rfcs/rfc3986.html>`__
+   - `rawurlencode <{+php-manual+}/rawurlencode>`__ in the PHP manual
+
+Replica Set Options
+~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Connection Option
+     - Description
+
+   * - :manual:`directConnection </reference/connection-string-options/#mongodb-urioption-urioption.directConnection>`
+     - | **Data Type**: {+bool-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = ['directConnection' => true];``
+       | **Connection URI Example**: ``directConnection=true``
+   
+   * - :manual:`replicaSet </reference/connection-string-options/#mongodb-urioption-urioption.replicaSet>`
+     - | **Data Type**: {+string-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = ['replicaSet' => 'replicaSetName'];``
+       | **Connection URI Example**: ``replicaSet=replicaSetName``
+   
+Connection Options
+~~~~~~~~~~~~~~~~~~
+
+TLS Options
+```````````
+To learn about the TLS options available in the {+driver-short+}, see the
+:ref:`TLS <php-tls>` page.
+
+Timeout Options
+```````````````
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Connection Option
+     - Description
+
+   * - :manual:`connectTimeoutMS </reference/connection-string-options/#mongodb-urioption-urioption.connecttimeoutms>`
+     - | **Data Type**: {+int-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = ['connectTimeoutMS' => 20000];``
+       | **Connection URI Example**: ``connectTimeoutMS=20000``
+
+   * - :manual:`socketTimeoutMS </reference/connection-string-options/#mongodb-urioption-urioption.sockettimeoutms>`
+     - | **Data Type**: {+int-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = ['socketTimeoutMS' => 400000];``
+       | **Connection URI Example**: ``socketTimeoutMS=400000``
```

Comment:
```suggestion
       | **MongoDB\\Client Example**: ``$uriOptions = ['socketTimeoutMS' => 20000];``
       | **Connection URI Example**: ``socketTimeoutMS=20000``
```

Likewise, this defaults to 300000 (five minutes). In practice, that actually exceeds PHP's own default [`max_execution_time`](https://www.php.net/manual/en/info.configuration.php#ini.max-execution-time) of 30 seconds, so it's likely PHP itself would kill the script before the driver gave up waiting for a server response.

Let's reduce this to 20 seconds just to demonstrate something more reasonable.